### DISCUSSION TITLE
I am confused how to add key to TreeView by the doc and the code when it comes to V2.

### DISCUSSION THREAD

**Discussion by Andras-Csanyi** [tui-cs/Terminal.Gui#3618] [Q&A] [answered]
Hey all,

Good to see V2 is coming around! Since the app is a new one I started with V2. I wanted to add multiple keystrokes to a TreeView, meaning if I press "a" it does this, if I press "b" another thing happens when an item is selected. The same what I described [here](https://github.com/gui-cs/Terminal.Gui/discussions/3589).

The doc says I should use `TreeView<T>.Keybindings.Add()` method where I can define what key executes what command. To define what happens when a command is executed I can define is by using the `TreeView<T>.AddCommand()` which is inherited from `View`, but it is not exposed. Shall I create my own type and inherit from `TreeView` and set the keystrokes up in the constructor? Do I understand correctly this was the plan?

**Comment by tznind** [comment, marked as answer]
I would continue to use the key event handler approach... keybinding is more for making standard functions remappable by end users and is primarily intended for new Types of view.

Can you link the bit of the v2 docs, maybe it could be cleared up?

**Reply by Andras-Csanyi** [reply]
I really appreciate this quick response. Here is the [doc](https://gui-cs.github.io/Terminal.GuiV2Docs/docs/migratingfromv1.html#updated-keyboard-api), and the confusing part if quoted below. It is highly probably I don't understand it fully as I don't know fully the language of this app.

```
The preferred way to handle single keystrokes is to use Key Bindings. Key Bindings map a key press to a [Command](https://gui-cs.github.io/Terminal.GuiV2Docs/api/Terminal.Gui.Command.html). A view can declare which commands it supports, and provide a lambda that implements the functionality of the command, using View.AddCommand(). Use the [View.Keybindings](https://gui-cs.github.io/Terminal.GuiV2Docs/api/Terminal.Gui.View.KeyBindings.html) to configure the key bindings.
```

**Reply by tznind** [reply]
Interesting, yes I can see the confusion.  I think that KeyBindings bit is written from the perspective of writting new View types in the core library rather than end applications but it could certainly be written better.

**Reply by tig** [reply]
> Interesting, yes I can see the confusion. I think that KeyBindings bit is written from the perspective of writting new View types in the core library rather than end applications but it could certainly be written better.

I agree. My bad.

However, I think this calls out that we need a KeyBinding-like extensibility mechanism such that apps don't have to resort to overrides to modify what keys are bound to what.
